pybossa/exporter/consensus_exporter.py

```diff
@@ -75,6 +75,39 @@ def _flatten(_obj, current_level, _prefix):
     _flatten(obj, 0, prefix)
     return flattened
 
+def get_contributor_answer(data, path, answer_field_config):
+
+    def match_nested_value(record):
+        match = [str(record[k]) == str(v) for k, v in key_value_pair.items()]
+        return all(match)
+
+    if not answer_field_config or not path:
+        return None
+    paths = path.split('.')
+    if answer_field_config.get('type') == 'categorical_nested':
+        key_values = answer_field_config.get('config', {}).get('keyValues', [])
+
+        field_name = paths[0]
+        key = paths[-1]
+        values = paths[1 : -1]
+        key_value_pair = {key_values[i]: values[i] for i in range(len(key_values))}
+
+        field_data = data.get(field_name, [])
+        target_field_data = [d for d in field_data if match_nested_value(d)]
+        return target_field_data[0][key] if target_field_data else None
+    else:
```

Comment:
is following code to be executed only for config.type = "categorical" and not for regular fields that has neither categorical, nor categorical nested configured?

Reply:
categorical_nested cases are taken care by 
`if answer_field_config.get('type') == 'categorical_nested':`
'else' includes cases that  config.type = "categorical" or config.type = "freetext"

Reply:
ya, fine w/ if part covering nested config. i was wondering instead of current else block, would it make sense to have explicit checks as
```python
if answer_field_config.get('type') == 'categorical_nested':
...
elif answer_field_config.get('type') in ['categorical', 'freetext']:
...
else: return None
```

Reply:
agree, will update pr